### PR TITLE
Makes ObjectID implement jsonschema.Builder.

### PR DESCRIPTION
**Pull request by smyrman**

### CODE REVIEW

File: id_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package mongo_test
+
+import (
+	"regexp"
+	"testing"
+
+	"gopkg.in/mgo.v2/bson"
+
```
Comment (rs) [inline id_test.go]: Unnecessary blank lines.

File: id_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	mongo "github.com/rs/rest-layer-mongo"
+)
+
+const (
+	validObjectID   = "59a40602952dbd0001c3ffc9"
+	invalidObjectID = "59a40602952dbd0001c3ffc9f"
+)
+
+func TestObjectIDValidate(t *testing.T) {
+
```
Comment (rs) [inline id_test.go]: Unnecessary blank line.

File: id_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	v := &mongo.ObjectID{}
+
+	t.Run("validObjectID", func(t *testing.T) {
+		id, err := v.Validate(validObjectID)
+		assert.Equal(t, bson.ObjectIdHex(validObjectID), id, "v.Validate(validObjectID)")
```
Comment (rs) [inline id_test.go]: I'm trying to progressively move out of testify in favor of plain Go testing by not using it in new tests and converting any tests I modify.

Comment (smyrman) [reply id_test.go]: I am rewriting it now, but can I ask why?

One of the things I feel `testify` really gives us, is a consistent output format on errors. When not using any framework (and I have in the past), I have ended up implementing my own checker functions in order to get consistent output, e.g. when comparing byte arrays.

File: id_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		if nil != id {
+			t.Errorf("v.Validate(invalidObjectID):\n %v (expect) != %v (actual)", nil, id)
+		}
+		if err == nil {
+			t.Error("v.Validate(invalidObjectID):\n expected error, got nil")
+		}
+	})
+}
+
+func TestObjectIDJSONSchmea(t *testing.T) {
```
Comment (smyrman) [inline id_test.go]: You can compare this one to the `testify/assert` version and see which one you like best.

File: id_test.go (hunk 1 above)
Comment (smyrman) [reply id_test.go]: fixed (testify/assert removed)

File: id_test.go (hunk 3 above)
Comment (smyrman) [reply id_test.go]: ☝️  @rs, ping so this comment don't get lost.

**Comment by smyrman** [comment]
I will merge this now.

**Pull request merged**